yal/main.py

```diff
@@ -232,53 +236,334 @@ async def state_catch_all(self):
 
     async def state_sexual_health_literacy_assessment(self):
         self.save_metadata("assessment_name", "sexual_health_literacy")
-        self.save_metadata("assessment_end_state", "state_assessment_end")
+        self.save_metadata(
+            "assessment_end_state", "state_sexual_health_literacy_assessment_end"
+        )
         return await self.go_to_state(AssessmentApplication.START_STATE)
 
     async def state_locus_of_control_assessment(self):
         self.save_metadata("assessment_name", "locus_of_control")
-        self.save_metadata("assessment_end_state", "state_assessment_end")
+        self.save_metadata(
+            "assessment_end_state", "state_locus_of_control_assessment_end"
+        )
         return await self.go_to_state(AssessmentApplication.START_STATE)
 
+    async def state_locus_of_control_assessment_end(self):
+        return EndState(
+            self,
+            self._(
+                "/n".join(
+                    [
+                        "[persona_emoji] Thanks for answering all the questions.",
+                        "",
+                        "That's all for now.",
+                    ]
+                )
+            ),
+        )
+
     async def state_depression_and_anxiety_assessment(self):
         self.save_metadata("assessment_name", "depression_and_anxiety")
-        self.save_metadata("assessment_end_state", "state_assessment_end")
+        self.save_metadata(
+            "assessment_end_state", "state_depression_and_anxiety_assessment_end"
+        )
         return await self.go_to_state(AssessmentApplication.START_STATE)
 
+    async def state_depression_and_anxiety_assessment_end(self):
+        score = self.user.metadata.get("assessment_score", 0)
+        if score <= 2:
+            # score of 0-2 high risk
+            risk = "high_risk"
+        else:
+            # score of 3-5 low risk
+            risk = "low_risk"
+
+        msisdn = normalise_phonenumber(self.inbound.from_addr)
+        whatsapp_id = msisdn.lstrip(" + ")
+        data = {
+            "depression_and_anxiety_risk": risk,
+            "depression_and_anxiety_score": score,
+        }
+        error = await rapidpro.update_profile(whatsapp_id, data, self.user.metadata)
+        if error:
+            return await self.go_to_state("state_error")
+        return await self.go_to_state(
+            "state_depression_and_anxiety_assessment_risk_message"
+        )
+
+    async def state_depression_and_anxiety_assessment_risk_message(self):
+        questions = {
+            "high_risk": self._(
+                "\n".join(
+                    [
+                        "[persona_emoji]  Esih! Sounds like it's been a rough "
```

Comment:
spelling error in "Esih" ?

Reply:
Its like that on the miroboard 😂 correcting quick!